docs/rfcs/xxxx-integrations-in-operator.md

```diff
@@ -0,0 +1,234 @@
+    ServiceMonitor.
+* Update `GrafanaAgent` to discover `MetricsIntegration`s and run integrations.
+* Update `MetricsInstance` to discover `IntegrationMonitor`s and collect
+  metrics from integrations.
+
+## Architecture
+
+### Running integrations
+
+The new CRD, MetricsIntegration, will be used for supporting all metrics-based
+integrations today. A metrics-based integration is an integration that exposes
+metrics. These integrations have an always-available set of common
+configuration fields. We define MetricsIntegration as a CRD specific for
+metrics-based integrations to add guarantees that these common configuration
+fields can be used when necessary for implementation details.
+
+The spec of MetricsIntegration primarily revolves around three fields:
+
+* `name`: The name of the integration (e.g., `node_exporter`, `mysqld_exporter`)
+* `type`: The type of integration
+* `config`: A YAML string declaring the configuration for the integration.
+
+`type` is critical for determining how the integration should be deployed.
+There are three supported values for `type`:
+
+* `daemonset`: Declares that the `name` integration should be run on every Node in the
+  Kubernetes cluster. It is invalid to have a GrafanaAgent CRD discover more
+  than one `daemonset` integrations with the same `name` . Example integration:
+  `node_exporter`.
+
+* `singleton`: Declares that the `name` integration may only be run once per
+  deployment. It is invalid to have a GrafanaAgent CRD discover more than one
+  `singleton` integration with the same `name.` Example integration:
+  `statsd_exporter`.
+
+* `normal`: Declares that the integration may exist any number of times per
+  deployment. Example integration: `mysqld_exporter`. This is the default.
+
+Users must supply the appropriate `type` value for the integration they are
+deploying. Integrations will be documented with their type to assist users in
+defining MetricsIntegrations. Failure to use the proper `type` value may lead
+to failure to start Grafana Agent integration pods.
+
+> Example of a valid integration:
+>
+> ```yaml
+> apiVersion: monitoring.grafana.com/v1alpha1
+> kind: MetricsIntegration
+> metadata:
+>   name: mysql
+>   namespace: default
+> spec:
+>   name: mysqld_exporter
+>   type: normal
+>   config: |
+>     data_source_name: root@(mysql.default:3306)/
+>     disable_collectors: [slave_status]
+> ```
+
+GrafanaAgent will be updated to discover MetricsIntegrations as part of its
+resource hierarchy, alongside MetricsInstances and LogsInstances. During
+reconcile, the following Kubernetes objects will be deployed:
+
+* One DaemonSet and Secret for all `daemonset` integrations, if
+  present in the resource hierarchy.
+
+* One Deployment and Secret for all other integrations, if non-`daemonset`
+  integrations are present in the resource hierarchy.
+
+* One Service for all integrations, if any integration was present in the
```

Comment:
Is this one service for each integration? Or all MetricsIntegrations will share a service?

Reply:
There will be exactly one service created as part of reconciling a GrafanaAgent if its resource hierarchy contains at least one MetricsIntegration. I'll try to clear that up in the doc :)